### PR TITLE
Fix app e2e race condition. Add more logs verbosity

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Add logs for when the e2e infinite loops fail to stop.

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda

### CODE REVIEW

File: src/lightning_app/testing/testing.py [hunk 1]
```diff
@@ -387,47 +390,41 @@ def run_app_in_cloud(
...
-            ]
-            app = lit_apps[0]
-
+            app = _fetch_app_by_name(client, project_id, name)
             if app.status.phase == V1LightningappInstanceState.RUNNING:
+                print("App is running, continuing with testing...")
+                break
+            if app.status.phase == V1LightningappInstanceState.STOPPED:
+                print("App is stopped, continuing with testing...")
                 break
```
Comment (carmocca) [inline src/lightning_app/testing/testing.py:402]: Maybe this should return directly?

File: src/lightning_app/testing/testing.py [hunk 2]
```diff
@@ -366,68 +379,51 @@ def run_app_in_cloud(
...
-            ).lightningapps
-            if app.name == name
-        ]
-        app = lit_apps[0]
-        app_url = app.status.url
-        while True:
-            sleep(1)
-            resp = requests.get(app_url + "/openapi.json")
-            if resp.status_code == 200:
+            if app.status.phase == V1LightningappInstanceState.STOPPED:
```
Comment (ethanwharris) [inline src/lightning_app/testing/testing.py:410]: Maybe this should be:
```suggestion
            elif app.status.phase != V1LightningappInstanceState.PENDING:
```
If it isn't pending then it started at some point then either stopped or failed - either way we can break out of the loop

**Review by ethanwharris — APPROVED** [review]
Nice!

File: src/lightning_app/testing/testing.py [hunk 3]
```diff
@@ -366,68 +379,51 @@ def run_app_in_cloud(
...
+                # there's a race condition if the app goes from pending to running to stopped before we evaluate the
+                # condition above. avoid it by checking stopped explicitly
+                print("App is stopped, finished testing...")
                 break
```
Comment (ethanwharris) [inline src/lightning_app/testing/testing.py:414]: I think view page is already typed as optional so we can just set it to None here - if apps exit that quick then the tests can't be making any checks on the UI
```suggestion
                print("App is stopped, finished testing...")
                view_page = None
                break
```

Comment (carmocca) [reply src/lightning_app/testing/testing.py]: I created a new page before we start the loop. I believe it's okay to the same page instance each iteration

**Review by awaelchli — APPROVED** [review]

**Pull request merged**